pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3782,6 +3817,236 @@ func (prt PipelineRunTest) reconcileRun(namespace, pipelineRunName string, wantE
 	return reconciledRun, clients
 }
 
+func TestReconcile_RemotePipelineRef(t *testing.T) {
+	names.TestingSeed()
+
+	// Set up a fake registry to push an image to.
+	s := httptest.NewServer(registry.New())
+	defer s.Close()
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ref := u.Host + "/testreconcile_remotepipelineref"
+
+	prs := []*v1beta1.PipelineRun{
+		tb.PipelineRun("test-pipeline-run-success",
+			tb.PipelineRunNamespace("foo"),
+			tb.PipelineRunSpec("test-pipeline",
+				tb.PipelineRunPipelineRefBundle(ref),
+				tb.PipelineRunServiceAccountName("test-sa"),
+				tb.PipelineRunResourceBinding("git-repo", tb.PipelineResourceBindingRef("some-repo")),
+				tb.PipelineRunResourceBinding("best-image", tb.PipelineResourceBindingResourceSpec(
+					&v1alpha1.PipelineResourceSpec{
+						Type: v1alpha1.PipelineResourceTypeImage,
+						Params: []v1alpha1.ResourceParam{{
+							Name:  "url",
+							Value: "gcr.io/sven",
+						}},
+					},
+				)),
+				tb.PipelineRunParam("bar", "somethingmorefun"),
+			),
+		),
+	}
+	funParam := tb.PipelineTaskParam("foo", "somethingfun")
+	moreFunParam := tb.PipelineTaskParam("bar", "$(params.bar)")
+	templatedParam := tb.PipelineTaskParam("templatedparam", "$(inputs.workspace.$(params.rev-param))")
+	ps := tb.Pipeline("test-pipeline",
```

Comment:
I'd prefer not to add new usages of test builders if we can avoid it. I'm (slowly) trying to weed them out in favor of structs.

Reply:
Sure. Happy to change it though I am genuinely curious why we are moving away from builders. As the change shows, it is a bit tedious to write out the whole thing.